### PR TITLE
Add asyncio implementation

### PR DESCRIPTION
**Pull request by eavanvalkenburg**
Hi @vanstinator, I've been looking at improving the component for HA for raincloud, and noticed I needed to do some stuff in the package as well, most importantly in creating a asyncio version of the package which HA prefers to work with.

Have a look and let me know what you think!

In this initial version:
- Created aio folder with core, controller, faucet
- Made small changes to functions for updating since that fits better with the `await` pattern
- No testing yet
- I think there are major improvements to make by centralizing logic in a base class for core, controller and faucet and just having the specific IO pieces in two files, one for AIO, one for sync, but this most likely leads to some breaking changes in the existing sync version

### CODE REVIEW

File: raincloudy/controller.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def __init__(self, parent, controller_id, index, faucets=None):
         """
         self.attributes = {}
         self._parent = parent
-        self.home = parent.html['home']
+        self.home = parent.html["home"]
```
Comment (vanstinator) [inline raincloudy/controller.py:28]: Let's keep using single quotes for now to minimize diffs.

**Review by vanstinator — COMMENTED** [review]
At first glance it looks good. Definitely get some tests in here too before we merge. Super appreciate the work!

**Comment by eavanvalkenburg** [comment]
Yeah I'll work on the tests, since of the quotes and stuff is because I use black formatting...

**Review by vanstinator — APPROVED** [review]
Thank you for tackling this!

**Comment by eavanvalkenburg** [comment]
No problem @vanstinator, the hard work was already done! Can you create a new release (maybe pre-release) with this functionality?

**Comment by vanstinator** [comment]
Apologies for letting this drop off my radar. Fall gets busy and my sprinklers get packed for the winter. I'm starting to prep for spring and that includes dusting off the raincloud home assistant bits to make sure it all works yet 😅. If you're still in the raincloud ecosystem and interested in landing this I can start moving forward.

**Comment by eavanvalkenburg** [comment]
Still using it, so would love to improve on it!

**Comment by vanstinator** [comment]
I've set up a new Github Actions workflow and it'll automatically publish pre-release builds to pypi on commits to `master`. So you should be able to start pulling this down almost immediately. And when you're happy with the state of the async bits I can cut a full stable release

**Comment by eavanvalkenburg** [comment]
Thanks @vanstinator mine is also in storage now, so will pick this back up once spring arrives here!

**Pull request merged**